### PR TITLE
Remove further uses of Location::none

### PR DESCRIPTION
**Pull request by tautschnig**
We now only have 12 uses of `Location::none` left, all of which have no reasonable location to use.

Resolves: #136

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs [hunk 1]
```diff
@@ -38,8 +38,10 @@ impl<'tcx> GotocCtx<'tcx> {
             Operand::Copy(place) | Operand::Move(place) =>
             // TODO: move is an opportunity to poison/nondet the original memory.
             {
-                let projection =
-                    unwrap_or_return_codegen_unimplemented!(self, self.codegen_place_stable(place));
+                let projection = unwrap_or_return_codegen_unimplemented!(
+                    self,
+                    self.codegen_place_stable(place, Location::none())
```
Comment (feliperodri) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:43]: Why is it right to keep this place without location?

File: kani-compiler/src/codegen_cprover_gotoc/context/goto_ctx.rs [hunk 2]
```diff
@@ -283,13 +289,14 @@ impl<'tcx> GotocCtx<'tcx> {
     pub fn register_initializer(&mut self, var_name: &str, body: Stmt) -> &Symbol {
         let fn_name = Self::initializer_fn_name(var_name);
         let pretty_name = format!("{var_name}::init");
+        let loc = *body.location();
         self.ensure(&fn_name, |_tcx, _| {
             Symbol::function(
                 &fn_name,
                 Type::code(vec![], Type::constructor()),
-                Some(Stmt::block(vec![body], Location::none())), //TODO is this block needed?
+                Some(Stmt::block(vec![body], loc)), //TODO is this block needed?
```
Comment (feliperodri) [inline kani-compiler/src/codegen_cprover_gotoc/context/goto_ctx.rs:297]: Not your fault, but this comment is odd. Do we have a GitHub issue to place it here?

File: tests/cargo-kani/simple-extern/src/lib.rs [hunk 3]
```diff
@@ -2,6 +2,8 @@
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 pub mod externs;
 pub use externs::external_c_assertion;
+// TODO: our reachability analysis does not see through C functions
+pub use externs::rust_add1;
```
Comment (feliperodri) [inline tests/cargo-kani/simple-extern/src/lib.rs:6]: Should we track this in a GitHub issue?

**Review by feliperodri — APPROVED** [review]

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs (hunk 1 above)
Comment (tautschnig) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:43]: Constants in operands have a `span`, but places don't. I don't know whether this is an intentional design in stable MIR or just an omission. @celinval?

File: tests/cargo-kani/simple-extern/src/lib.rs (hunk 3 above)
Comment (tautschnig) [reply tests/cargo-kani/simple-extern/src/lib.rs:6]: https://github.com/model-checking/kani/issues/3263

File: kani-compiler/src/codegen_cprover_gotoc/context/goto_ctx.rs (hunk 2 above)
Comment (tautschnig) [reply kani-compiler/src/codegen_cprover_gotoc/context/goto_ctx.rs:297]: Reviewed all uses, it's actually indeed not necessary. Added an assertion to make sure this remains ok.

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs (hunk 1 above)
Comment (tautschnig) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:43]: Added a TODO comment.

**Pull request merged**